### PR TITLE
feature: Add configurable labels for Argo Workflows

### LINKED ISSUE #2780: Enhancement: Allow Argo Label customization
If we could add labels to argo workflows, we could more easily group workflows in the Argo UI. 

Current state:
I see that here the labels are only being built using `_base_labels`: https://github.com/Netflix/metaflow/blob/13a7cad8081aff1f34db8d4e663d8940e65ebbdc/metaflow/plugins/argo/argo_workflows.py#L892 
However, `_base_labels` is only being built from `_base_kubernetes_labels`: https://github.com/Netflix/metaflow/blob/13a7cad8081aff1f34db8d4e663d8940e65ebbdc/metaflow/plugins/argo/argo_workflows.py#L189C34-L189C57
Since `_base_kubernetes_labels` is effectively hardcoded, there's no customization possible in the labels, i see also that there's a TODO exactly what i would suggest here, that there is an environment variable to use here: 
https://github.com/Netflix/metaflow/blob/13a7cad8081aff1f34db8d4e663d8940e65ebbdc/metaflow/plugins/argo/argo_workflows.py#L402-L409

Proposed Solution:
Add `METAFLOW_ARGO_WORKFLOWS_LABELS`. Then use it to set the labels.

Example PR: https://github.com/Netflix/metaflow/pull/2781

### PR DESCRIPTION
**Pull request by Capiru**
## Summary

Adds `METAFLOW_ARGO_WORKFLOWS_LABELS` for configuring labels on Argo WorkflowTemplates and generated Workflow resources.

Custom labels are intentionally not applied to Pods, JobSets, or Sensors. Step-level Kubernetes labels remain controlled by `METAFLOW_KUBERNETES_LABELS` and `@kubernetes(labels=...)`.

Metaflow-owned labels take precedence over configured labels.

## Usage

```shell
export METAFLOW_ARGO_WORKFLOWS_LABELS="team=ml-platform,cost-center=12345,environment=production"
python myflow.py argo-workflows create
```

Configured labels are added to:

- `WorkflowTemplate.metadata.labels`
- `WorkflowTemplate.spec.workflowMetadata.labels`, which supplies labels to generated Workflow resources

## Testing

- Tests default and configured labels
- Tests protection of Metaflow-owned labels
- Tests invalid configuration
- Verifies labels in compiled WorkflowTemplate JSON

Fixes #2780

### CODE REVIEW

File: metaflow/plugins/argo/argo_workflows.py [hunk 1]
```diff
@@ -188,7 +193,7 @@ def __init__(
         self.triggers, self.trigger_options = self._process_triggers()
         self._schedule, self._timezone = self._get_schedule()
 
-        self._base_labels = self._base_kubernetes_labels()
+        self._base_labels = self._base_argo_labels()
```
Comment (saikonen) [inline metaflow/plugins/argo/argo_workflows.py]: just to double-check. base_labels also applies to sensors. Is it intended that the configurable `ARGO_WORKFLOWS_LABELS` applies to all argo resources, not only workflow/workflow templates?

File: metaflow/plugins/argo/argo_workflows.py [hunk 2]
```diff
@@ -403,10 +408,26 @@ def trigger(cls, name, parameters=None):
...
+        Format: comma-separated key=value pairs (e.g., "team=ml,env=prod")
+        """
+        labels = self._base_kubernetes_labels()
+
+        if ARGO_WORKFLOWS_LABELS:
+            env_labels = parse_kube_keyvalue_list(
+                ARGO_WORKFLOWS_LABELS.split(","), requires_both=True
+            )
+            validate_kube_labels(env_labels)
+            labels.update(env_labels)
```
[the remark this answers is not in the document]
Comment (saikonen) [reply metaflow/plugins/argo/argo_workflows.py]: this is relevant. Internal labels should be protected against overwriting

[the remark this answers is not in the document]
Comment (saikonen) [reply metaflow/plugins/argo/argo_workflows.py]: step-specific labels should override global ones through env vars.

File: metaflow/plugins/argo/argo_workflows.py (hunk 1 above)
Comment (Capiru) [reply metaflow/plugins/argo/argo_workflows.py]: This is a nice catch, I made it explicit so that it only applies to Workflow/Templates

**Review by Shriprasad-P — COMMENTED** [review]
## Review

**PR:** feature: Add configurable labels for Argo Workflows

**Touched:** `metaflow/metaflow_config.py`, `metaflow/plugins/argo/argo_workflows.py`, `metaflow/plugins/kubernetes/kube_utils.py`, `test/unit/test_argo_workflows_labels.py`, `test/ux/core/test_argo_compilation.py`

- Addresses a concrete correctness/reliability issue based on the title and diff.
- Security-sensitive change — please double-check edge cases and defaults.
- CI/tooling change — confirm the pipeline still passes on this branch.
- Size looks manageable (+193/-14).

Commenting as a drive-by reviewer after reading the diff. Happy to look again if maintainers want a deeper pass on a specific file.
